Add default theme

rushenn opened the pull request:
Added the following as part of the default theming:

- SCSS partical files
- New HTML page structure
- Popup previews for images
- Syntax highlighting for codes
- Mobile Support
- SVG icons for buttons and similar elements 
- 'Vendors' directory for all 3rd part vendors

**Note**: Active state is not working in both links and preview contents. This has to be fixed in the next PR


![screen shot 2018-02-26 at 12 29 48](https://user-images.githubusercontent.com/13393018/36657002-ca20256e-1af0-11e8-8405-3b8541bce46b.png)

rushenn commented:
@deekoder ^

deekoder commented:
I am testing and reviewing. Will complete today.

deekoder commented:
Let me leave my feedback so far  more later : 
 * This can be the minio template and not the default template. Will also run this question by AB.  
*  The chinese/japanese dropdowns need to have links. We need to write some code to plot these so that for each english doc, there is a corresponding lang doc. The other option is not to do the dropdown inside gluegun and simply have links for each language site somewhere else (inside the text for that doc as a heading).
* SourcePro looks nice on the body of the doc. But its elongating the capitalized text too much. The Main Minio text needs to be in Mark font. 

Will check with designers for more feedback.
More later.

deekoder commented:
@rushenn  lets have a call about this PR.

kaankabalak commented:
Looks so good @rushenn, great work! Just a very minor issue, when the language is set to Japanese, the text and the dropdown icon look a bit crammed to me. However, Chinese and English look good enough. Depending on how it would look for other options, maybe we can make the width of the `select` just a bit longer to get more space. You can refer to the image below for further reference.

<img width="1280" alt="screen shot 2018-02-26 at 6 05 06 pm" src="https://user-images.githubusercontent.com/27314853/36706823-83197000-1b20-11e8-81f6-da91292d43a3.png">

rushenn commented:
@deekoder 
> This can be the minio template and not the default template. Will also run this question by AB.
We can have this as the default theme and give color customization such as text color, bg color etc. in order to customize for Minio theme. 

> The Main Minio text needs to be in Mark font.
It's logo (It should be an imahe) . Currently I'm using our official wordtext logo.  

I'll address the rest issues soon.

rushenn commented:
@kaankabalak I just noticed this. Select has fixed width based on the first option selected. I think I should try a custom dropdown option here. I'll check it out.

deekoder commented:
We spoke yesterday about this PR @rushenn  I will hold approval till those changes are done whenever you do them.

rushenn commented:
@deekoder @kaankabalak I have updated the UI and fixed the dropdown issue. Please review.

![gluegun-new](https://user-images.githubusercontent.com/13393018/37764592-1d02d6d2-2de8-11e8-8e11-a646b1a48bac.png)

kaankabalak commented:
Also, the docs with broken links still seem to be displayed on the sidebar when they should be hidden instead.

kaankabalak commented on lib/_sidebar.erb:
```diff
@@ -1,57 +1,35 @@
+            <img src="https://minio.io/img/logo/wordmark_white.svg" alt="">
+        </a>
+
+        <ul class="doc-link">
```
I think we should check for `@site_map['Documents']` not being nil here, as we might get an error if the user doesn't have anything listed for Documents in the .yml file.

kaankabalak commented on lib/_sidebar.erb:
```diff
@@ -1,57 +1,35 @@
+                                <%- else -%>
+                                    <%- active_keyword="" -%>
+                                <%- end -%>
+                                <li><a class="<%=active_keyword%>" href="#<%=key2["Slug"]%>"><%=key2.keys[0]%></a></li>
```
We would have to check for `reveal(raw(key2["Link"]))` to be true to prevent documents that have broken links to appear on the sidebar.

kaankabalak commented on lib/_content.erb:
```diff
@@ -0,0 +1,20 @@
+                                                <%- active_keyword="" -%>
+                                        <%- end -%>
+
+                                        <div class="doc-preview__item <%=active_keyword%>" id="<%=key2["Slug"]%>">
```
We would once again have to check for `reveal(raw(key2["Link"]))` to be true to prevent the divs of the documents that have broken links to appear on the html file.

deekoder commented:
I will wait for these feedback from @kaankabalak  to resolve and then continue the review.

rushenn commented:
@kaankabalak @deekoder Fixed the images and vendors rendering issue. 

Since this PR is related to UI, can we merge this and address the validation related issues in a new PR?

deekoder commented:
Ok sounds good. Let me validate all the UX elements and open a new issue.

deekoder commented:
closing this for rebased one #43
